src/components/EditFilter/EditFilter.jsx

```diff
@@ -1,49 +1,28 @@
+const { connect } = require('react-redux')
 const React = require('react')
-const Filter = require('../../models/Filter')
 const FilterSuggester = require('../FilterSuggester')
 const hookUpStickyNav = require('../hookUpStickyNav')
 
 class EditFilter extends React.Component {
   constructor(props) {
     super(props)
-    const { filter } = props
-    this.state = {
-      valueHasError: false,
-      key: filter,
-      value: new Filter(filter).retrieve(),
-    }
+    this.state = { valueHasError: false }
   }
 
   save(event) {
     event.preventDefault()
-    if (this.state.value.length < 1) {
+    if (this.props.filter.query.length < 1) {
       this.setState({ valueHasError: true })
       return
     }
     this.setState({ valueHasError: false })
-    let key = this.state.key.trim()
-    if (key.length < 1) {
-      key = this.state.value
-    }
-    const filter = new Filter(key)
-    filter.store(this.state.value)
-    if (this.props.filter !== key) {
-      this.props.delete(this.props.filter)
-    }
-    this.props.save(key)
+    this.props.dispatch({ type: 'FILTERS_UPDATE', filter: this.props.filter })
```

Comment:
Since it is only used twice, I think this is ok for now. I did take your advice and do this in the render method, though.